### PR TITLE
[AIRFLOW-XXX] Update to new logo

### PR DESCRIPTION
**Pull request by ashb**
I have removed the static/favicon.ico as this wasn't referenced in the
templates, and wasn't configured to be served as `/favicon.ico` (only as
`/static/favicon.ico`) so has not been in use since 51a38a in 2014.

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by kaxil** [comment]
The only thing missing is probably a **large transparent Airflow logo**. 

We have a transparent image as emoji but not with bigger size.

**Comment by ashb** [comment]
> The only thing missing is probably a **large transparent Airflow logo**.
> 
> We have a transparent image as emoji but not with bigger size.

Done.

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
From CI:
```
diff --git a/airflow/www/templates/airflow/circles.html b/airflow/www/templates/airflow/circles.html
index b200508..0a9ea27 100644
--- a/airflow/www/templates/airflow/circles.html
+++ b/airflow/www/templates/airflow/circles.html
@@ -1,4 +1,6 @@
 {#
+ TODO: This license is not consistent with license used in the project.
+       Delete the inconsistent license and above line and rerun pre-commit to insert a good license.
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
```

**Comment by ashb** [comment]
Curious....

**Comment by feluelle** [comment]
What do you guys think of adding the `wordmark` logo to the [landscape](https://github.com/cncf/landscape) ?

**Comment by ashb** [comment]
Happy to if they accept it - do we need to be "Cloud Native" (what ever the _hell_ that means) for it to be accepted?

**Comment by feluelle** [comment]
From their [Trailmap](https://github.com/cncf/landscape#trail-map):
Screenshot 2019-09-20 at 15 55 18

File: docs/index.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 
 
-.. image:: img/pin_large.png
+.. image:: ../airflow/www/static/pin_large.png
```
Comment (ashb) [inline docs/index.rst:21]: .... I think this should fail to build the docs - I've removed that file. Why isn't it?

Comment (ashb) [reply docs/index.rst:21]: Oh no, I just made it a slightly smaller file.

**Pull request merged**